packages/lisk-p2p/src/p2p.ts

```diff
@@ -49,40 +49,75 @@ const BASE_10_RADIX = 10;
 
 export class P2P extends EventEmitter {
 	private readonly _config: P2PConfig;
-	private readonly _peerPool: PeerPool;
 	private readonly _httpServer: Server;
-	private readonly _scServer: SCServerUpdated; // Until we get comeplete definition for SCServer
+	private _isActive: boolean;
 	private readonly _newPeers: Set<PeerInfo>;
-	// TODO ASAP: private readonly _triedPeers: Set<PeerInfo>;
-	private _nodeStatus: P2PNodeStatus;
+	private readonly _triedPeers: Set<PeerInfo>;
+
+	private _nodeInfo: P2PNodeInfo;
+	private readonly _peerPool: PeerPool;
+	private readonly _scServer: SCServerUpdated;
 
 	public constructor(config: P2PConfig) {
 		super();
 		this._config = config;
-		this._nodeStatus = {
+		this._nodeInfo = {
 			wsPort: config.wsPort,
 			os: platform(),
 			version: config.version,
 			height: 0,
 		};
+		this._isActive = false;
 		this._newPeers = new Set();
-		// TODO ASAP: this._triedPeers = new Set();
+		this._triedPeers = new Set();
 
 		this._peerPool = new PeerPool();
 		this._httpServer = http.createServer();
 		this._scServer = attach(this._httpServer) as SCServerUpdated;
 	}
 
+	public get isActive(): boolean {
+		return this._isActive;
+	}
+
+	public get nodeInfo(): P2PNodeInfo {
+		return this._nodeInfo;
+	}
+
+	/**
+	 * This is not a declared as a setter because this method will need
+	 * invoke an async RPC on Peers to give them our new node status.
+	 */
+	public applyNodeInfo(nodeInfo: P2PNodeInfo): void {
+		this._nodeInfo = nodeInfo;
+		const peerList = this._peerPool.getAllPeers();
+		peerList.forEach(peer => {
+			peer.applyNodeInfo(nodeInfo);
+		});
+	}
+
 	/* tslint:disable:next-line: prefer-function-over-method */
 	public applyPenalty(penalty: P2PPenalty): void {
 		penalty;
 	}
 
-	// TODO ASAP: public getNetworkStatus(): NetworkStatus {};
+	public getNetworkStatus(): P2PNetworkStatus {
+		const newPeers = [...this._newPeers.values()];
+		const triedPeers = [...this._triedPeers.values()];
+		const availablePeers = newPeers.concat(triedPeers);
```

Comment:
is `availablePeers` are all the peers that are available but don't have any connection yet? or, does it include all the peers available or known including the connected peers? If its the former case then I think we need to filter `newPeers` and `triedPeers` to remove peers that we are already connected to and then concat.

Reply:
Since the `getNetworkStatus` call will be used by integration tests, it's important that it shows the internal state of the node as closely as possible.

I think it's more important that this interface be detailed rather than user-friendly. Users of the method can do their own sanitisation of the raw data.

Integration test cases will want to check the state of the _newPeers and _triedPeers maps explicitly (since it's a core concept in the LIPs), so we need the name of the exposed properties to reflect the name internal properties for which we want to allow introspection. We could call it `newPeersList` instead of `newPeers` maybe?

The less sanitisation/formatting we do on our end, the more performant this call will be.

Reply:
@shuse2 What is your opinion about this?

Reply:
I think without `availablePeers` is fine.
`availablePeers` seems ambiguous in this case because it will include all the peers regardless of connection since `triedPeer` will include the peer with outbound connection.
Also, not really useful if it's just combination of `newPeers` and `triedPeers`, user can just combine by themselves?

Reply:
In case we want to have a concatenated list of `newPeers` and `triedPeers` and if we are exposing them then we don't need `availablePeers` list as its redundant or the user can make it out by themselves. I was thinking that network status is just about knowing (connected and unconnected Peers) but since `newPeers` and `triedPeers` will be used in the network test then its a better to add it. I guess in future we will need a separate class for handling peer bucket or peer directory (like I discussed once) that includes newPeer/triedPeer/bannedPeer and then it will be easier to test these lists.